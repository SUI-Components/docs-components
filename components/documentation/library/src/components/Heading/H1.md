# H1
> version undefined

HTML `h1` element DOC styled

## Installation
`npm install @s-ui/documentation-library`

## Usage
```scss
// include H1 SASS module
@include '@s-ui/documentation-library/lib/components/Heading/Heading.scss'
```

```js
// import H1 DOC component
import { H1 } from '@s-ui/documentation-library/lib/components/Heading/Heading.js'
```

or

```scss
// include all SASS modules
@include '@s-ui/documentation-library/lib/index.scss'
```

```js
// import all DOC components you need
import { H1 } from '@s-ui/documentation-library'
```

## Prop-Types

| Name | Description | Values  | Default | Required |
| ---: |:---:| ---:| ---: |:---: |
| elementType | undefined | | |  [ ]  |
