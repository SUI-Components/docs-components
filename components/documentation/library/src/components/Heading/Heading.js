/* eslint react/prop-types: 0 */
import React from 'react'
import PropTypes from 'prop-types'
import cx from 'classnames'

import Base from '../Base.core'

export const HEADING_ELEMENT = {
  h1: {
    elementType: 'h1',
    className: (cn, mode) =>
      cx(
        'sui-studio-doc-heading',
        'sui-studio-doc-heading-h1',
        {
          [`sui-studio-doc-heading-mode-${mode}`]: mode
        },
        cn
      )
  },
  h2: {
    elementType: 'h2',
    className: (cn, mode) =>
      cx(
        'sui-studio-doc-heading',
        'sui-studio-doc-heading-h2',
        {
          [`sui-studio-doc-heading-mode-${mode}`]: mode
        },
        cn
      )
  },
  h3: {
    elementType: 'h3',
    className: (cn, mode) =>
      cx(
        'sui-studio-doc-heading',
        'sui-studio-doc-heading-h3',
        {
          [`sui-studio-doc-heading-mode-${mode}`]: mode
        },
        cn
      )
  },
  h4: {
    elementType: 'h4',
    className: (cn, mode) =>
      cx(
        'sui-studio-doc-heading',
        'sui-studio-doc-heading-h4',
        {
          [`sui-studio-doc-heading-mode-${mode}`]: mode
        },
        cn
      )
  }
}

/**
 * Create HTML tag titles.
 */
const Heading = ({elementType, children, ...props}) => (
  <Base {...props} elementType={elementType}>
    {children}
  </Base>
)

Heading.displayName = 'Heading'
Heading.propTypes = {}
Heading.defaultProps = {}

Heading.propTypes = {
  /**
   * HTML tag of the title
   **/
  elementType: PropTypes.oneOf(Object.keys(HEADING_ELEMENT)).isRequired
}

/**
 * HTML `h1` element DOC styled
 */
const H1 = ({
  children,
  className,
  elementType = HEADING_ELEMENT.h1.elementType,
  mode,
  ...props
}) => (
  <Heading
    {...props}
    className={HEADING_ELEMENT.h1.className(className, mode)}
    elementType={elementType}
    mode={mode}
  >
    {children}
  </Heading>
)
H1.displayName = 'H1'
H1.propTypes = {}
H1.defaultProps = {}

/**
 * HTML `h1` element DOC styled
 */
const H2 = ({
  children,
  className,
  elementType = HEADING_ELEMENT.h2.elementType,
  mode,
  ...props
}) => (
  <Heading
    {...props}
    className={HEADING_ELEMENT.h2.className(className, mode)}
    elementType={elementType}
    mode={mode}
  >
    {children}
  </Heading>
)
H2.displayName = 'H2'
H2.propTypes = {}
H2.defaultProps = {}

/**
 * HTML `h3` element DOC styled
 */
const H3 = ({
  children,
  className,
  elementType = HEADING_ELEMENT.h3.elementType,
  mode,
  ...props
}) => (
  <Heading
    {...props}
    className={HEADING_ELEMENT.h3.className(className, mode)}
    elementType={elementType}
    mode={mode}
  >
    {children}
  </Heading>
)

H3.displayName = 'H3'
H3.propTypes = {}
H3.defaultProps = {}

/**
 * HTML `h4` element DOC styled
 */
const H4 = ({
  children,
  className,
  elementType = HEADING_ELEMENT.h4.elementType,
  mode,
  ...props
}) => (
  <Heading
    {...props}
    className={HEADING_ELEMENT.h4.className(className, mode)}
    elementType={elementType}
    mode={mode}
  >
    {children}
  </Heading>
)

H4.displayName = 'H4'
H4.propTypes = {}
H4.defaultProps = {}

export default Heading

export {Heading, H1, H2, H3, H4}
