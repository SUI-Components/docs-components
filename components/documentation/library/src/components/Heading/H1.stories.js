/* eslint react/prop-types: 0 */
import React from 'react'

import {H1} from './Heading'

export default {
  title: 'Components/Heading/H1',
  component: H1,
  argTypes: {
    mode: {
      control: {
        type: 'select',
        options: {
          undefined: undefined,
          light: 'light',
          dark: 'dark'
        }
      }
    }
  }
}

const Template = props => <H1 {...props} />

export const Default = Template.bind({})

Default.args = {
  children: 'H1'
}
