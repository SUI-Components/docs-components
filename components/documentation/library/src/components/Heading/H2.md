# H2
> version undefined

HTML `h1` element DOC styled

## Installation
`npm install @s-ui/documentation-library`

## Usage
```scss
// include H2 SASS module
@include '@s-ui/documentation-library/lib/components/Heading/Heading.scss'
```

```js
// import H2 DOC component
import { H2 } from '@s-ui/documentation-library/lib/components/Heading/Heading.js'
```

or

```scss
// include all SASS modules
@include '@s-ui/documentation-library/lib/index.scss'
```

```js
// import all DOC components you need
import { H2 } from '@s-ui/documentation-library'
```

## Prop-Types

| Name | Description | Values  | Default | Required |
| ---: |:---:| ---:| ---: |:---: |
| elementType | undefined | | |  [ ]  |
