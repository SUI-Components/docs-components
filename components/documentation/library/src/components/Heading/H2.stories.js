/* eslint react/prop-types: 0 */
import React from 'react'

import {H2} from './Heading'

export default {
  title: 'Components/Heading/H2',
  component: H2,
  argTypes: {
    mode: {
      control: {
        type: 'select',
        options: {
          undefined: undefined,
          light: 'light',
          dark: 'dark'
        }
      }
    }
  }
}

const Template = props => <H2 {...props} />

export const Default = Template.bind({})

Default.args = {
  children: 'H2'
}
