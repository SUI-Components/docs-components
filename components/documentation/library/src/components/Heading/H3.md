# H3
> version undefined

HTML `h3` element DOC styled

## Installation
`npm install @s-ui/documentation-library`

## Usage
```scss
// include H3 SASS module
@include '@s-ui/documentation-library/lib/components/Heading/Heading.scss'
```

```js
// import H3 DOC component
import { H3 } from '@s-ui/documentation-library/lib/components/Heading/Heading.js'
```

or

```scss
// include all SASS modules
@include '@s-ui/documentation-library/lib/index.scss'
```

```js
// import all DOC components you need
import { H3 } from '@s-ui/documentation-library'
```

## Prop-Types

| Name | Description | Values  | Default | Required |
| ---: |:---:| ---:| ---: |:---: |
| elementType | undefined | | |  [ ]  |
