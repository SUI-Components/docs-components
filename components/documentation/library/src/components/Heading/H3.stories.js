/* eslint react/prop-types: 0 */
import React from 'react'

import {H3} from './Heading'

export default {
  title: 'Components/Heading/H3',
  component: H3,
  argTypes: {
    mode: {
      control: {
        type: 'select',
        options: {
          undefined: undefined,
          light: 'light',
          dark: 'dark'
        }
      }
    }
  }
}

const Template = props => <H3 {...props} />

export const Default = Template.bind({})

Default.args = {
  children: 'H3'
}
