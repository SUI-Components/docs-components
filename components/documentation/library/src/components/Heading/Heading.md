# Heading
> version undefined

Create HTML tag titles.

## Installation
`npm install @s-ui/documentation-library`

## Usage
```scss
// include Heading SASS module
@include '@s-ui/documentation-library/lib/components/Heading/Heading.scss'
```

```js
// import Heading DOC component
import { Heading } from '@s-ui/documentation-library/lib/components/Heading/Heading.js'
```

or

```scss
// include all SASS modules
@include '@s-ui/documentation-library/lib/index.scss'
```

```js
// import all DOC components you need
import { Heading } from '@s-ui/documentation-library'
```

## Prop-Types

| Name | Description | Values  | Default | Required |
| ---: |:---:| ---:| ---: |:---: |
| elementType | HTML tag of the title | | |  [x]  |
