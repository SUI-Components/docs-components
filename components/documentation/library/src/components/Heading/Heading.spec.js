/* global expect */
import React, {createRef} from 'react'
import ReactDOM from 'react-dom'

import {Heading} from './Heading'

import setupEnvironment from '../../helpers/setupEnvironment'

describe('components', () => {
  describe('Heading', () => {
    describe('default', () => {
      const Component = Heading
      const setup = setupEnvironment(Component)

      test('should render without crashing', () => {
        // Given
        const props = {
          elementType: 'h1'
        }

        // When
        const component = <Component {...props} />

        // Then
        const div = document.createElement('div')
        ReactDOM.render(component, div)
        ReactDOM.unmountComponentAtNode(div)
      })

      test('should NOT render null', () => {
        // Given
        const props = {
          elementType: 'h1'
        }

        // When
        const {container} = setup(props)

        // Then
        expect(container.innerHTML).toBeString()
        expect(container.innerHTML).not.toHaveLength(0)
      })

      test('should have propTypes and defaultProps ', () => {
        // Given
        const component = Component

        // When

        // Then
        expect(component.hasOwnProperty('propTypes')).toBeTruthy()
        expect(component.hasOwnProperty('defaultProps')).toBeTruthy()
      })

      test('should have proper displayName attribute', () => {
        // Given
        // const props = {}

        // When
        const component = Component

        // Then
        expect(component.displayName).toBeString()
        expect(component.displayName).not.toHaveLength(0)
        expect(component.displayName).toMatchSnapshot()
      })

      describe('forwardRef', () => {
        test('should have reference', () => {
          // Given
          const props = {elementType: 'h1'}
          const ref = createRef()

          // When
          const component = <Component {...props} ref={ref} />
          const div = document.createElement('div')
          ReactDOM.render(component, div)

          // Then
          expect(ref.current).not.toBe(undefined)
          expect(ref.current.nodeName).toBe('H1')
        })

        test('should have defined elementType reference', () => {
          // Given
          const props = {elementType: 'h3'}
          const ref = createRef()

          // When
          const component = <Component {...props} ref={ref} />
          const div = document.createElement('div')
          ReactDOM.render(component, div)

          // Then
          expect(ref.current).not.toBe(undefined)
          expect(ref.current.nodeName).toBe('H3')
        })
      })
    })
  })
})
