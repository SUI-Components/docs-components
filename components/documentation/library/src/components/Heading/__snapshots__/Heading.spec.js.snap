// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading default should have proper displayName attribute 1`] = `"Heading"`;
